wolfssl/wolfcrypt/cpuid.h

```diff
@@ -80,23 +90,38 @@
     #define CPUID_SM4         0x0080    /* SM4 enc/dec */
     #define CPUID_SB          0x0100    /* Speculation barrier */
 
-    #define IS_AARCH64_AES(f)       ((f) & CPUID_AES)
-    #define IS_AARCH64_PMULL(f)     ((f) & CPUID_PMULL)
-    #define IS_AARCH64_SHA256(f)    ((f) & CPUID_SHA256)
-    #define IS_AARCH64_SHA512(f)    ((f) & CPUID_SHA512)
-    #define IS_AARCH64_RDM(f)       ((f) & CPUID_RDM)
-    #define IS_AARCH64_SHA3(f)      ((f) & CPUID_SHA3)
-    #define IS_AARCH64_SM3(f)       ((f) & CPUID_SM3)
-    #define IS_AARCH64_SM4(f)       ((f) & CPUID_SM4)
-    #define IS_AARCH64_SB(f)        ((f) & CPUID_SB)
+    #define IS_AARCH64_AES(f)       (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_AES)
+    #define IS_AARCH64_PMULL(f)     (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_PMULL)
+    #define IS_AARCH64_SHA256(f)    (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_SHA256)
+    #define IS_AARCH64_SHA512(f)    (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_SHA512)
+    #define IS_AARCH64_RDM(f)       (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_RDM)
+    #define IS_AARCH64_SHA3(f)      (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_SHA3)
+    #define IS_AARCH64_SM3(f)       (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_SM3)
+    #define IS_AARCH64_SM4(f)       (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_SM4)
+    #define IS_AARCH64_SB(f)        (WOLFSSL_ATOMIC_COERCE_UINT(f) & CPUID_SB)
 
 #endif
 
 #ifdef HAVE_CPUID
-    void cpuid_set_flags(void);
     word32 cpuid_get_flags(void);
 
-    /* Public APIs to modify flags. */
+    static WC_INLINE int cpuid_get_flags_ex
+    #if defined(WOLFSSL_ATOMIC_OPS) && !defined(SINGLE_THREADED)
+        (wolfSSL_Atomic_Uint *flags)
+    #else
+        (word32 *flags)
+    #endif
```

Comment:
Please no macro gurads in func definition. Put this logic in a generic flags type.